cw-orch/src/interchain/interchain_channel.rs

```diff
@@ -0,0 +1,317 @@
+// This struct is used to create and/or track the state of a channel between two chains. 
+// This is very modular to be able to follow transactions, channel creation...
+
+use base64::Engine;
+use crate::follow_ibc_execution::AckResponse;
+use base64::engine::general_purpose;
+use tokio::time::{sleep, Duration};
+use tonic::transport::Channel;
+
+use crate::{CosmTxResponse, DaemonError};
+use crate::daemon::queriers::DaemonQuerier;
+use crate::queriers::Node;
+
+#[derive(Debug, Clone)]
+pub struct TxId {
+	pub chain_id: String,
+	pub channel: Channel,
+	pub tx_hash: String
+}
+
+#[derive(Debug, Clone)]
+pub struct InterchainPort{
+	pub chain: Channel,
+	pub chain_id: String,
+	pub port: String,
+	pub channel: Option<String>
+}
+
+#[derive(Debug)]
+pub struct InterchainChannel{
```

Comment:
This new struct allows one to query a lot of information in an IBC channel.
Based on events, it can query transactions corresponding to packet transfers and channel creation.
It is mostly used for now for debugging IBC channel creation and transactions. 
For now, this is better than the IBCTracker struct or the queriers::Ibc struct because it can query whole transactions (useful to follow execution). However, it might be a good thing to merge that to the existing IBCTracker and/or querier::Ibc structs